frontend/src/utils/websockets.js

```diff
@@ -43,25 +44,35 @@ export function connectWS(topic, onMessageCallback) {
 
   const connect = () => {
     let reconnectCounter = 0;
+    let badGatewayStatus = false;
```

Comment:
why not having `errorType` (string) instead of the boolean?

Reply:
errorType is a string (the one passed to the noConnection action). 

badGatewayStatus on the other hand is just a flag in here

Reply:
that's obvious for me too.

What i am trying to suggest is, instead of having that flag, which IMHO it's pointless,

have the `connectionErrorType`
which will be initialized with the corresponding connection error type.
i.e. BAD_GATEWAY_ERROR, NO_CONNECTION_ERROR

What would happen if u would have more errors in the future? will u introduce a flag for each of them?

Reply:
to introduce and use `let connectionErrorType` instead of that badGatewayStatus flag.

Reply:
got it. Thanks for explaining me. Pushed the changes